ui/panels-plugin/src/plugins/time-series-chart/TimeSeriesChartPanel.tsx

```diff
@@ -216,9 +209,15 @@ export function TimeSeriesChartPanel(props: TimeSeriesChartProps) {
 
         if (showTimeSeries) {
           timeSeriesMapping.push(
-            getTimeSeries(seriesId, seriesIndex, formattedSeriesName, visual, timeScale, seriesColor)
+            getTimeSeries(seriesId, timeChartData.length, formattedSeriesName, visual, timeScale, seriesColor)
```

Comment:
Oh wait, I think I see, `timeChartData.length` represents the current index of the visible series, since we'll push each visible series into that array one by one

Reply:
^ i had the same confusion initially. Maybe we should update this to use `seriesIndex++` for readability or add a comment here why `timeChartData.length` is used?

Reply:
Yeah agreed, I’ll refactor to use seriesIndex instead, I tried with that first but where seriesIndex++ is called needs to change

Reply:
The current `seriesIndex` can't be used since it's also needed for looping through the color palette, if I try to use it and move `seriesIndex++` inside the `if (showTimeSeries)` condition, all the colors end up being the same after a legend item is selected, ex: 
<img width="835" alt="image" src="https://github.com/perses/perses/assets/9369625/6d7f2e5e-9ccc-45fb-b4ed-43114d41140b">

Reply:
Renamed some variables so it's more clear (we now have a datasetIndex var) and added comments!